### PR TITLE
PHOENIX-7381 : Skip last_ddl_timestamp validation when UCF is defined on table and cache entry is not old

### PR DESCRIPTION
**Pull request by palashc**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java [hunk 1]
```diff
@@ -1812,6 +1812,47 @@ public void testCacheUpdatedBeforeDDLOperations() throws Exception {
...
+                upsert(conn1, tableName, true);
+            }
+
+            MetricsMetadataCachingSource.MetadataCachingMetricValues newMetricValues
+                    = metricsSource.getCurrentMetricValues();
+
+            assertEquals("There should have been no timestamp validation requests.",
+                    0,
+                    newMetricValues.getValidateDDLTimestampRequestsCount()
+                            - oldMetricValues.getValidateDDLTimestampRequestsCount());
```
Comment (virajjasani) [inline phoenix-core/src/it/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java:1853]: `newMetricValues.getValidateDDLTimestampRequestsCount()` should also be 0?

Comment (palashc) [reply phoenix-core/src/it/java/org/apache/phoenix/cache/ServerMetadataCacheIT.java:1853]: There will be requests from other tests, that's why I am checking the difference before and after just this test.

**Comment by palashc** [comment]
@virajjasani Does this change look good? If yes, we can get it in.

File: phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java [hunk 2]
```diff
@@ -810,12 +810,10 @@ private boolean avoidRpcToGetTable(boolean alwaysHitServer, Long resolvedTimesta
...
             //QueryOptimizer needs that to decide whether the index can be used
-            if (overrideUcfToDefault) {
+            if (overrideUcfToAlways) {
                 effectiveUpdateCacheFreq =
-                    (Long) ConnectionProperty.UPDATE_CACHE_FREQUENCY.getValue(
-                        connection.getQueryServices().getProps()
-                            .get(QueryServices.DEFAULT_UPDATE_CACHE_FREQUENCY_ATRRIB));
-                ucfInfoForLogging = "connection-level-default";
+                    (Long) ConnectionProperty.UPDATE_CACHE_FREQUENCY.getValue("ALWAYS");
+                ucfInfoForLogging = "index-level-override";
```
Comment (virajjasani) [inline phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java:816]: nit: for the logging purpose, let's use `override-to-always` or something similar

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/UCFWithDisabledIndexWithDDLValidationIT.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ private static void initCluster() throws Exception {
         Map<String, String> props = Maps.newConcurrentMap();
         props.put(BaseScannerRegionObserverConstants.PHOENIX_MAX_LOOKBACK_AGE_CONF_KEY,
                 Integer.toString(60 * 60 * 1000));
-        props.put(QueryServices.DEFAULT_UPDATE_CACHE_FREQUENCY_ATRRIB, "ALWAYS");
+        props.put(QueryServices.DEFAULT_UPDATE_CACHE_FREQUENCY_ATRRIB, "NEVER");
```
Comment (virajjasani) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/UCFWithDisabledIndexWithDDLValidationIT.java:43]: do we want to test with both `ALWAYS` and `NEVER`?

**Review by virajjasani — COMMENTED** [review]
Few nits, +1 otherwise

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/UCFWithDisabledIndexWithDDLValidationIT.java (hunk 3 above)
Comment (palashc) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/UCFWithDisabledIndexWithDDLValidationIT.java:43]: @virajjasani With DDL timestamp validation, UCF should not be ALWAYS, so I think it is okay. WDYT?

File: phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java (hunk 2 above)
Comment (palashc) [reply phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java]: done

**Review by virajjasani — APPROVED** [review]
+1, pending jenkins build results

**Pull request merged**